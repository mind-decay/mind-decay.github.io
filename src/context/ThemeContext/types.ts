import { Nullable } from 'types';

export type ThemeType = 'light' | 'dark';

export type ThemeContextType = {
  theme: ThemeType;
  toggleTheme: () => void;
};

export const isThemeType = (theme: Nullable<string>): theme is ThemeType => {
  return typeof theme === 'string' && ['light', 'dark'].includes(theme);
};
