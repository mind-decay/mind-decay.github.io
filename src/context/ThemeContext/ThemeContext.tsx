import { ReactNode, createContext, useContext, useEffect, useState } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';

import { ThemeContextType, ThemeType, isThemeType } from './types';
import { Nullable } from 'types/helpers';

import { darkTheme, lightTheme } from 'styles/themes';

const ThemeContext = createContext<Nullable<ThemeContextType>>(null);

export const useTheme = (): ThemeContextType => {
  const context = useContext(ThemeContext);

  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }

  return context;
};

type ThemeProviderProps = {
  children: ReactNode;
};

export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const [theme, setTheme] = useState<ThemeType>('dark');
  const isLightTheme = theme === 'light';

  const toggleTheme = () => {
    const updatedTheme = isLightTheme ? 'dark' : 'light';
    setTheme(updatedTheme);
    localStorage.setItem('theme', updatedTheme);
  };

  useEffect(() => {
    const savedTheme = localStorage.getItem('theme');
    const prefersLight = window.matchMedia && window.matchMedia('(prefers-color-scheme: light)').matches;
    const checkCurrentTheme = isThemeType(savedTheme);

    if (checkCurrentTheme) {
      setTheme(savedTheme);
    } else if (prefersLight) {
      setTheme('light');
    }
  }, []);

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      <StyledThemeProvider theme={isLightTheme ? lightTheme : darkTheme}>{children}</StyledThemeProvider>
    </ThemeContext.Provider>
  );
};
