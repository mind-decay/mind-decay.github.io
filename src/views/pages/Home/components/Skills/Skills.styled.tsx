import styled from 'styled-components';
import { screenBreakpoints } from 'styles/utils';

export const SkillsContainer = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  row-gap: 5.5rem;
  width: 100%;
  margin: 10rem 0;

  @media screen and (max-width: ${screenBreakpoints.tablet}px) {
    row-gap: 2rem;
    margin: 4rem 0;
  }
`;
