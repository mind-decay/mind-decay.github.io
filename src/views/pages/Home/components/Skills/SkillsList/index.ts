export * from './SkillsList';
export * from './SkillsList.styled';
