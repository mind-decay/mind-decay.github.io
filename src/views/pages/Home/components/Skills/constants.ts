export const skillsData = [
  'HTML',
  'CSS',
  'SCSS',
  'Styled-Components',
  'Tailwindcss',
  'JavaScript',
  'TypeScript',
  'React',
  'Next.js',
  'Solid.js',
  'Redux',
  'Effector',
  'Webpack',
  'Vite',
  'Rollup',
  'Gulp',
  'Jest',
  'React Testing Library',
  'Cypress',
  'Playwright',
  'REST API',
  'Formik',
  'React Hook Form',
  'Axios',
  'AntD',
  'MUI',
  'Chakra UI',
];
