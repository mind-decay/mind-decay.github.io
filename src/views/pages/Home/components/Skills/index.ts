export * from './Skills';
export * from './Skills.styled';
