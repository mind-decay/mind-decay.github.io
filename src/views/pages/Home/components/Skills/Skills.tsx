import { Title } from 'views/components/Typography';
import { SkillsContainer } from './Skills.styled';
import { SkillsList } from './SkillsList';

export const skillsData = [
  'HTML',
  'CSS',
  'SCSS',
  'Styled-Components',
  'Tailwindcss',
  'JavaScript',
  'TypeScript',
  'React',
  'Next.js',
  'Solid.js',
  'Redux',
  'Effector',
  'Webpack',
  'Vite',
  'Rollup',
  'Gulp',
  'Jest',
  'React Testing Library',
  'Cypress',
  'Playwright',
  'REST API',
  'Formik',
  'React Hook Form',
  'Axios',
  'AntD',
  'MUI',
  'Chakra UI',
];

export const Skills = () => {
  return (
    <SkillsContainer id="skills">
      <Title variant="h2" gradient="mainGradient">
        Навыки
      </Title>

      <SkillsList skillsData={skillsData} />
    </SkillsContainer>
  );
};
