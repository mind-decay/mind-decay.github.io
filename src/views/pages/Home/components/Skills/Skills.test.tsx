import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { ThemeProvider } from 'context/ThemeContext';

import { Skills } from './Skills';

test('renders Skills section', () => {
  const { container } = render(
    <ThemeProvider>
      <Skills />
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});
