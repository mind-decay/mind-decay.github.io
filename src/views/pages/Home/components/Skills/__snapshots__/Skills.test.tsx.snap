// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Skills section 1`] = `
<div>
  <section
    class="sc-jTQCzO bJvUwC"
    id="skills"
  >
    <h2
      class="sc-guDLey dFIBZG"
    >
      Навыки
    </h2>
    <ul
      class="sc-gLLuof dpOfjX"
    >
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          HTML
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          CSS
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          SCSS
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Styled-Components
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Tailwindcss
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          JavaScript
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          TypeScript
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          React
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Next.js
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Solid.js
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Redux
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Effector
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Webpack
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Vite
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Rollup
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Gulp
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Jest
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          React Testing Library
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Cypress
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Playwright
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          REST API
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Formik
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          React Hook Form
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Axios
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          AntD
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          MUI
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Chakra UI
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Reatom
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Storybook
        </span>
      </li>
      <li
        class="sc-iBdnpw shfkT"
      >
        <span>
          Gravity UI
        </span>
      </li>
    </ul>
  </section>
</div>
`;
